src/twisted/test/test_defer.py

```diff
@@ -11,8 +11,17 @@
 import gc, traceback
 import re
 
+from twisted.python.compat import _PY3, _PY34PLUS
+
+if _PY34PLUS:
+    from asyncio import new_event_loop, Future, CancelledError
+    asyncSkip = None
+else:
+    asyncSkip = "asyncio not available before python 3.4"
+
```

Comment:
I've gone back and forth over this throughout the years, so I'm sympathetic to that point of view, but my current thinking goes like this:

We know that `asyncio` is part of the stdlib on 3.4+.  It _must_ be importable there or there's some kind of major problem with the python installation.  In the case where we expect it to be available because the interpreter says it should be, we go ahead and import it, and fail hard if something is so messed up that it's not importable.  In the case where we _don't_ think it should be available, there's no point in trying.

For _application_ code there's a benefit to being lenient and trying to function in the face of some ambiguity, but for _tests_ this still strikes me as the better way of dealing with conditional imports.

I'm open to being convinced otherwise though.